src/Node/HTTP2.purs

```diff
@@ -0,0 +1,310 @@
+-- | Bindings to the [*Node.js* HTTP/2](https://nodejs.org/docs/latest/api/http2.html) API.
+-- |
+-- | Requires *Node.js v14.17.0* (for
+-- | [`AbortController`](https://nodejs.org/docs/latest/api/globals.html#class-abortcontroller))
+-- |
+-- | ## `Effect` event-callback asynchronous API
+-- |
+-- | The __Node.HTTP2.Client__ and __Node.HTTP2.Server__ modules provide a
+-- | low-level `Effect` event-callback API which is a thin FFI wrapper
+-- | around the __*Node.js* HTTP/2__ types and functions.
+-- |
+-- | ## `Aff` asynchronous API
+-- |
+-- | The __Node.HTTP2.Client.Aff__ and __Node.HTTP2.Server.Aff__ modules provide
+-- | a high-level asynchronous `Aff` API so that
+-- | [“you don’t even have to think about callbacks.”](https://github.com/purescript-contrib/purescript-aff/tree/main/docs#escaping-callback-hell)
+-- | That’s nice because when we attach callbacks to every possible event
+-- | and then handle the events, it’s hard to keep track of the order
+-- | in which events are occuring, and what context we’re in when an event
+-- | handler is called.
+-- |
+-- | With the `Aff` API we can write HTTP/2 clients and services in plain flat
+-- | one-thing-after-another
+-- | effect style. But network peers are perverse and they may not
+-- | send things to us in the order in which we expect. So we will want to
+-- | reintroduce some of the indeterminacy that we get from an event-callback
+-- | API. That’s what the
+-- | [`Parallel ParAff Aff`](https://pursuit.purescript.org/packages/purescript-aff/docs/Effect.Aff#t:ParAff)
+-- | instance is for.
+-- | We can use the functions in
+-- | [`Control.Parallel`](https://pursuit.purescript.org/packages/purescript-parallel/docs/Control.Parallel)
+-- | to run `Aff` effects concurrently.
```

Comment:
I don't think this is the place to write a small tutorial on `Aff`. I think that is better served by people clicking on the `Aff` type and reading any docs on that type.

Reply:
> I don't think this is the place to write a small tutorial on `Aff`.

Beginners are going to need this explanation. Maybe I should move this into the `docs` directory?

Reply:
I've learned there is no `docs` directory for any of the __purescript-node__ repos. I’d rather decide whether to merge this PR and then move the docs around later.

Reply:
I agree with @JordanMartinez that this isn't the right place for an `Aff` tutorial, but it's a fine place to concretely demonstrate using this module via `Aff`.

Reply:
Meaning, I think these docs should be moved to `Aff` specifically, and then the module header above may say, "Refer to `Aff` for more details. Here's a few examples of what you likely want." `Aff` isn't well-documented in the first place aside from Nate's video on it and my learning repo which summarizes Nate's video in text/code.

Reply:
> I think these docs should be moved to `Aff` specifically, and then the module header above may say, "Refer to `Aff` for more details. Here's a few examples of what you likely want." 

I love that idea but let's do it after this PR.